src/body.js

```diff
@@ -50,10 +50,10 @@ export default class Body {
 			body = Buffer.from(body.buffer, body.byteOffset, body.byteLength);
 		} else if (body instanceof Stream) {
 			// Body is stream
-		} else if (isFormData(body)) {
-			// Body is an instance of formdata-node
-			boundary = `nodefetchformdataboundary${getBoundary()}`;
-			body = Stream.Readable.from(formDataIterator(body, boundary));
+		} else if (body instanceof FormData) {
+			// Body is an spec compatible FormData
+			body = formDataToBlob(body);
+			boundary = body.type.split('=')[1];
```

Comment:
Do we even need a reference to `boundary`? Your `formDataToBlob` returns a `Blob` with type `multipart/form-data boundary=%boundary%`, so I think it can be just reused later.

Reply:
yea, kind of do... if the header class would be created first and passed down to the Body class (like how [github/fetch dose it](https://github.com/github/fetch/blob/master/fetch.js#L217-L259)) then i could assign the `content-type` but it's not possible the way it's structured right now.

both deno and undici do something like a util function that [extract body and contentType](https://fetch.spec.whatwg.org/#concept-bodyinit-extract) "10. Return body and Content-Type."
Unlike ourself where we extend a Body class. which is the "wrong thing" to do.

for instances `Response.prototype.arrayBuffer === Request.prototype.arrayBuffer` is only true in node-fetch 
not in undici, Deno or the browser is the function the same thing. Looking more into how [undici](https://github.com/nodejs/undici/blob/main/lib/fetch/body.js#L25) & [Deno](https://github.com/jimmywarting/deno/blob/2d519e9c5beb54e38c354ce65f22ecffbefdbc2f/extensions/fetch/22_body.js#L309) have implemented it more according to spec then i realize now that we have mostly done: "yea, we can do it this way and it will roughly do the same thing" so we have not followed the spec correctly